crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -168,7 +169,18 @@ impl RestartManagerActor {
                 }
             }
             Err(err) => {
-                error!("Fail to read tedge-agent state: {err}");
+                match err {
+                    StateError::FromIo(e) => match e.kind() {
+                        std::io::ErrorKind::NotFound => {
+                            info!("tedge-agent state file is not present")
```

Comment:
Indeed, not an error.

I would even consider to log nothing in this case. This just means that no restart operation has ever be requested on this device.

Reply:
Though it might offer more insight to see if the agent is loading state from an existing file or not (when troubleshooting issues where you are not sure if state is being loaded or not which controls what actions are done)